server.js

```diff
@@ -4458,6 +4458,75 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Bitbucket Pipelines integration.
+camp.route(/^\/bitbucket\/pipelines\/([^/]+)\/([^/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var user = match[1];  // eg, atlassian
+  var repo = match[2];  // eg, adf-builder-javascript
+  var branch = match[3] || 'master';  // eg, development
+  var format = match[4];
+  var apiUrl = 'https://api.bitbucket.org/2.0/repositories/'
+    + encodeURIComponent(user) + '/' + encodeURIComponent(repo)
+    + '/pipelines/?fields=values.state&page=1&pagelen=2&sort=-created_on'
+    + '&target.ref_type=BRANCH&target.ref_name=' + encodeURIComponent(branch);
+
+  var badgeData = getBadgeData('build', data);
+
+  request(apiUrl, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
```

Comment:
This block here and the later sections where you're checking for `if (res.statusCode !== 200)` and `if (res.statusCode === 404)` are really common so we have now added a [helper](https://github.com/badges/shields/blob/3cb71037fdbed20f74a18880ed9edad4273a08c5/lib/error-helper.js). You can simplify all that to:

https://github.com/badges/shields/blob/75503ec5144b349ab42c4877b774d3acac8157f3/server.js#L6515-L6518

Reply:
Great, I'll use that. I'll make the changes later today.